src/allocator/memserver_allocator.cpp

```diff
@@ -532,24 +568,172 @@ void Memserver_Allocator::restore(uint64_t destRegionId, uint64_t destOffset,
 }
 
 Fam_Backup_Info Memserver_Allocator::get_backup_info(std::string BackupName) {
-    ostringstream message;
     struct stat sb;
     Fam_Backup_Info info;
-    info.name = (char *)BackupName.c_str();
+    info.bname = (char *)BackupName.c_str();
+    info.size = -1;
+    info.uid = -1;
+    info.gid = -1;
+    info.mode = -1;
     if (stat(BackupName.c_str(), &sb) == -1) {
-        info.size = -1;
-        info.uid = -1;
-        info.gid = -1;
-        info.mode = -1;
-    } else {
-        info.size = sb.st_size;
-        info.uid = sb.st_uid;
-        info.gid = sb.st_gid;
-        info.mode = sb.st_mode;
+	return info;
+    }
+    string metafile = BackupName + ".info";
+    int fileid = open((char *)metafile.c_str(), O_RDONLY, S_IRWXU | S_IRWXG | S_IRWXO);
+    if (fileid == -1) {
+            cout << "Memserver_Allocator::get_backup_info - metafile doesn't exist" << endl;
+            return info;
+    }
+    long pgsz = sysconf(_SC_PAGESIZE);
+    lseek(fileid, pgsz - 1, SEEK_SET);
+    char *srcaddr;
+    srcaddr =
+        (char *)mmap(NULL, pgsz, PROT_READ, MAP_SHARED, fileid, 0);
+
+    if (srcaddr == MAP_FAILED) {
+       cout << "Memserver_Allocator::get_backup_info - mmap of backup meta info failed.";
+       return info;
+    }
+    char *metainfo = (char *)malloc(pgsz);
+    memcpy(metainfo, srcaddr, pgsz);
+    munmap(srcaddr, pgsz);
+    close(fileid);
+
+    std::stringstream buffer;
+    buffer.str(metainfo);
+    std::string dataitemName;
+      
+for (std::string token; std::getline(buffer, token); ) {
+        size_t pos = token.find(":");
+        std::string attribute;
+        std::string value;
+
+           if( pos <= token.length() ) {
+                attribute = token.substr(0,pos);
+                value = token.substr(pos+1, token.length());
+           }
+                if ( attribute.compare("dataitemName" ) == 0) {
+                        dataitemName = value;
+		}
+                if ( attribute.compare("dataitemSize" ) == 0) {
+                        info.size = std::stol(value);
+		}
+
+                if ( attribute.compare("uid" ) == 0)
+                        info.uid = std::stoi(value);
+
+                if ( attribute.compare("gid" ) == 0)
+                        info.gid = std::stoi(value);
+
+                if ( attribute.compare("mode" ) == 0)
+                        info.mode = std::stoi(value);
+
     }
     return info;
 }
 
+std::string get_backup_metadata(string metafile) {
```

Comment:
is get_backup_info and get_backup_metadata same ? Should we use get_backup_info and read the information as structure.